test/e2e/replica_set/replica_set_test.go

```diff
@@ -24,6 +24,7 @@ func TestReplicaSet(t *testing.T) {
 	t.Run("Create MongoDB Resource", mongodbtests.CreateMongoDBResource(&mdb, ctx))
 	t.Run("Config Map Was Correctly Created", mongodbtests.AutomationConfigConfigMapExists(&mdb))
 	t.Run("Stateful Set Reaches Ready State", mongodbtests.StatefulSetIsReady(&mdb))
+	t.Run("Stateful Set has OwnerReference", mongodbtests.StatefulSetHasOwnerReference(&mdb))
```

Comment:
I believe we have a "BasicConnectivity" or "BasicFunctionality" test that incorporates a lot of the common things we test frequently. This new test would be a great one to add there!